lib/ecto/migrator.ex

```diff
@@ -105,13 +120,27 @@ defmodule Ecto.Migrator do
       attempt(repo, module, :forward, :down, :down, opts)
         || attempt(repo, module, :backward, :change, :down, opts)
         || raise Ecto.MigrationError, "#{inspect module} does not implement a `down/0` or `change/0` function"
-      verbose_schema_migration repo, "update schema migrations", fn ->
-        SchemaMigration.down(repo, version, opts[:prefix])
-      end
+    end
+
+    verbose_schema_migration repo, "update schema migrations", fn ->
+      SchemaMigration.down(repo, version, opts[:prefix])
     end
   end
 
   defp run_maybe_in_transaction(repo, module, fun) do
+    Task.async(fn ->
+      do_run_maybe_in_transaction(repo, module, fun)
+    end)
+    |> Task.await(:infinity)
```

Comment:
It is better to use `Task.yield` instead of rescuing inside the Task but you would probably need a Task.Supervisor so you can use `async_nolink`.

Reply:
Or alternatively we just let the task crash and bring the link down. The only concern is if the error report will be too verbose or not.

Reply:
Let's go with let it crash first and then we can review the logged output after this is merged. :)